File: java/codingame/practice/easy/Descent/Player.java
import java.util.*;
import java.io.*;
import java.math.*;

/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/
class Player {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int[] mountainHeight = new int[8];
        // game loop
        while (true) {
            for (int i = 0; i < 8; i++) {
                mountainHeight[i] = in.nextInt();
            }

            int max = Arrays.stream(mountainHeight).max().getAsInt();
            int maxIndex = indexOf(mountainHeight, max);
            // Write an action using System.out.println()
            // To debug: System.err.println("Debug messages...");

            System.out.println(maxIndex); // The index of the mountain to fire on.
        }
    }

    public static int indexOf(int[] array, int value){
        int index = -1;
        int i = 0;
        for(int elem:array){
            if(elem == value) {
                index = i;
                break;
            }
            i++;
        }
        return index;
    }
}

File: scala/codingame/practice/easy/the_descent/src/main/scala/Player.scala
import math._
import scala.util._

/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/
object Player extends App {

    // game loop
    while(true) {
        // Write an action using println
        // To debug: Console.err.println("Debug messages...")
        val mountains = for (i <- 0 until 8) yield (i,readInt)
        println(mountains.sortWith(_._2 > _._2).head._1.toString) // The index of the mountain to fire on.
    }
}